modules/services/owncloud-client.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  options = {
+    services.owncloud-client = {
+      enable = mkEnableOption "Owncloud Client";
+    };
+  };
+
+  config = mkIf config.services.owncloud-client.enable {
+    systemd.user.services.owncloud-client = {
```

Comment:
If this should be started in a graphical session, then I think you also need

```
Install = {
  WantedBy = [ "graphical-session.target" ];
};
```
here.